### PR TITLE
Create & Use npm path nudge

### PR DESCRIPTION
**Pull request by DomT4**
Various things:

* Stops `npm`-using formulae just breaking completely when `HOMEBREW_SANDBOX` is set, and spamming up the `$HOME` cache/config when it isn't.
* Stops `npm`-using formulae being reliant on `post_install` which currently means we can't bottle the most significant part of each of these compiles.
* Sets a standard format for how to get `npm` back in the `$PATH`, since `superenv` removes it and the answer so far has been to re-add `HOMEBREW_PREFIX/"bin"` manually which risks picking up unexpected deps, or using `post_install` which isn't great UX.

### CODE REVIEW

**Comment by DomT4** [comment]
CC @Homebrew/owners @bfontaine

**Comment by DomT4** [comment]
Would also impact #40517 and #39771

File: Library/Formula/skinny.rb [hunk 1]
```diff
@@ -1,5 +1,3 @@
-require "formula"
-
 class UniversalNpm < Requirement
```
Comment (DomT4) [inline Library/Formula/skinny.rb]: I'm not completely sure why this is a requirement given we ship Node `--with-npm` by default. IMO we should scrap the block and requirement completely.

File: Library/Formula/tegh.rb [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
+
+  head "https://github.com/D1plo1d/tegh.git", :branch => "develop"
 
-  depends_on 'node'
+  depends_on "node"
 
   def install
+    if build.head?
+      ENV.prepend_path "PATH", "#{Formula["node"].opt_libexec}/npm/bin"
+      ENV["HOME"] = ".brew_home"
```
Comment (mikemcquaid) [inline Library/Formula/tegh.rb]: Probably wants to be a full path with e.g. `buildpath` in it. Otherwise :+1: on the PR

**Pull request closed without merging**